superset/databases/filters.py

```diff
@@ -51,3 +56,56 @@ def apply(self, query: Query, value: Any) -> Query:
                 ),
             )
         )
+
+
+class DatabaseUploadEnabledFilter(BaseFilter):
+    """
+    Custom filter for the GET list that filters all databases based on allow_file_upload
+    """
+
+    name = _("Upload Enabled")
+    arg_name = "upload_is_enabled"
+
+    def can_access_databases(  # noqa pylint: disable=no-self-use
+        self,
+        view_menu_name: str,
+    ) -> Set[str]:
+        return {
+            security_manager.unpack_database_and_schema(vm).database
+            for vm in security_manager.user_view_menu_names(view_menu_name)
+        }
+
+    def apply(self, query: Query, value: Any) -> Query:
+
+        filtered_query = query.filter(Database.allow_file_upload)
+
+        database_perms = security_manager.user_view_menu_names("database_access")
+        schema_access_databases = self.can_access_databases("schema_access")
+
+        datasource_access_databases = self.can_access_databases("datasource_access")
+
+        if hasattr(g, "user"):
+            allowed_schemas = [
+                app.config["ALLOWED_USER_CSV_SCHEMA_FUNC"](db, g.user)
+                for db in datasource_access_databases
+            ]
+
+            if len(allowed_schemas):
+                return filtered_query
+
+        schema_filtered_query = filtered_query.filter(
```

Comment:
Reading this in context now, we could probably reuse the filtered_query var here instead of creating a new one. The new var helps explain what's being added, but then it's reused again below in the perm filters, so it's not consistent. Adding some comments could be another option.